File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        differenceMap = {}
        for i in range(len(nums)):
            difference = target - nums[i]

            if difference not in differenceMap:
                differenceMap[nums[i]] = i
            else:
                answer = [differenceMap[difference], i]
                return answer

        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int begin = 0;
        int end = numbers.size()-1;
        while (begin < end) {
            if (numbers[begin] + numbers[end] == target){
                return {begin+1, end+1};
            } else if (numbers[begin] + numbers[end] < target){
                begin++;
            } else{
                end--;
            }
        }
        return {begin+1, end+1}; 
    }
};
